presidio-analyzer/analyzer/recognizer_result.py

```diff
@@ -1,15 +1,29 @@
+from . import AnalysisExplanation
+
+
 class RecognizerResult:
 
-    def __init__(self, entity_type, start, end, score):
+    def __init__(self, entity_type, start, end, score,
+                 analysis_explanation: AnalysisExplanation = None):
```

Comment:
Consider adding a field for textual explanations that will be used onlly if the analysis_explanation is None (only one of the two could be given). This would make life easier for recognizer developers that don't know how to use the AnalysisExplanation class, but want to provide some textual information about how their recognizer decided on an output. 
Something like:

`

     def __init__(self, entity_type, start, end, score, analysis_explanation: AnalysisExplanation=None, textual_result_explanation: str=None):

        if analysis_explantion and textual_result_explanation:
            raise ValueError("Expecting either analysis_explanation or textual_result_explanation, not both")
    
        if not analysis_explanation:
            analysis_explanation = AnalysisExplanation(free_text=textual_result_explanation)
`

Reply:
Actually, this is not a good solution, because you can create a new instance of `AnalysisExplanation` in `RecognizerResult`: you don't have in hand the parameters needed to instantiate it. I think it's a good enough at this point to accept the `AnalysisExplanation` as is, and have more verbose doc string of how to add simple text

Reply:
I'm not sure what would be best. In other languages we would do overloading. In python the closest solution is to add more parameters. Let's take it offline and decide.